cpp/src/merge/merge.cu

```diff
@@ -0,0 +1,381 @@
+        temp_columns_to_free.push_back(gdf_col_ptr(new gdf_column{}, gdf_col_deleter));
+        gdf_column * new_right_column_ptr = temp_columns_to_free.back().get();
+
+        *new_left_column_ptr = allocate_like(*left_col, RETAIN, stream);
+        if (new_left_column_ptr->valid) {
+            CUDA_TRY( cudaMemcpyAsync(new_left_column_ptr->valid, left_col->valid, sizeof(cudf::valid_type)*gdf_num_bitmask_elements(left_col->size), cudaMemcpyDefault, stream) );
+            new_left_column_ptr->null_count = left_col->null_count;
+        }
+        
+        *new_right_column_ptr = allocate_like(*right_col, RETAIN, stream);
+        if (new_right_column_ptr->valid) {
+            CUDA_TRY( cudaMemcpyAsync(new_right_column_ptr->valid, right_col->valid, sizeof(cudf::valid_type)*gdf_num_bitmask_elements(right_col->size), cudaMemcpyDefault, stream) );
+            new_right_column_ptr->null_count = right_col->null_count;
+        }
+
+        gdf_column * tmp_arr_input[2] = {left_col, right_col};
+        gdf_column * tmp_arr_output[2] = {new_left_column_ptr, new_right_column_ptr};
+        CUDF_TRY( sync_column_categories(tmp_arr_input, tmp_arr_output, 2) );
+
+        left_cols_sync[i] = new_left_column_ptr;
+        right_cols_sync[i] = new_right_column_ptr;
+    }
+
+    table left_sync_table(left_cols_sync);
+    table right_sync_table(right_cols_sync);
+
+    std::vector<gdf_column*> left_key_cols_vect(key_cols.size());
+    std::transform(key_cols.cbegin(), key_cols.cend(), left_key_cols_vect.begin(),
+                  [&] (cudf::size_type const index) { return left_sync_table.get_column(index); });
+    
+    std::vector<gdf_column*> right_key_cols_vect(key_cols.size());
+    std::transform(key_cols.cbegin(), key_cols.cend(), right_key_cols_vect.begin(),
+                  [&] (cudf::size_type const index) { return right_sync_table.get_column(index); });
+
+    auto left_key_table = device_table::create(left_key_cols_vect.size(), left_key_cols_vect.data());
+    auto right_key_table = device_table::create(right_key_cols_vect.size(), right_key_cols_vect.data());
+    rmm::device_vector<int8_t> asc_desc_d(asc_desc);
+
+    rmm::device_vector<index_type> merged_indices = generate_merged_indices(*left_key_table, *right_key_table, asc_desc_d, nulls_are_smallest, stream);
+
+    // Allocate output table
+    bool nullable = has_nulls(left_sync_table) || has_nulls(right_sync_table);
+    table destination_table(left_sync_table.num_rows() + right_sync_table.num_rows(),
+                            column_dtypes(left_sync_table),
+                            column_dtype_infos(left_sync_table),
+                            nullable, false, stream);
+    for (cudf::size_type i = 0; i < destination_table.num_columns(); i++) {
+        gdf_column const* left_col = left_sync_table.get_column(i);
+        gdf_column * out_col = destination_table.get_column(i);
+        
+        if (left_col->dtype != GDF_STRING_CATEGORY){
```

Comment:
NVCategory / GDF_STRING_CATEGORY are no more.

Reply:
> 
> 
> NVCategory / GDF_STRING_CATEGORY are no more.

OK. STRING is is then. Thanks.

Reply:
I think the whole logic has to be reimplemented to work on string columns. @davidwendt @jrhemstad can probably advise better.